checklist/cloud_build.yaml

```diff
@@ -0,0 +1,60 @@
+steps:
+  # The first two steps are required to identify and inject un-redacted secrets.
+  - name: gcr.io/cloud-builders/npm
+    args:
+      - install
+      - yaml
+  - name: gcr.io/cloud-builders/npm
+    args:
+      - run
+      - replace-secrets
+      - --
+      - checklist
+    secretEnv:
+      - PRIVATE_KEY
+      - WEBHOOK_SECRET
+
+  # Compile the app from Typescript into dist/.
+  - name: gcr.io/cloud-builders/npm
+    dir: checklist
+    args:
+      - install
+  - name: gcr.io/cloud-builders/npm
+    dir: checklist
+    args:
+      - run
+      - build
+
+  # Copy over the environment and package file.
+  - name: ubuntu
+    dir: checklist
+    args:
+      - cp
+      - .env
+      - dist/
+  - name: ubuntu
+    dir: checklist
+    args:
+      - cp
+      - package.json
+      - dist/
+
+  # Deploy the app.
+  - name: gcr.io/cloud-builders/gcloud
+    dir: checklist
+    args:
+      - functions
+      - deploy
+      - --source
+      - dist/
+      - checklist-bot
```

Comment:
Unsure about these names or this GCP deployment process.

Reply:
https://github.com/ampproject/amp-github-apps/blob/master/DEPLOYMENT.md

Reply:
The name here is the name of the Cloud Function, which can be whatever as long as it stays constant. This should be fine. See above for guidance on automating the deployment with Cloud Build and for guidance on setting up the shared keyring/encrypted secrets.